source/common/http/header_map_impl.cc

```diff
@@ -19,10 +19,15 @@ constexpr size_t MinDynamicCapacity{32};
 // This includes the NULL (StringUtil::itoa technically only needs 21).
 constexpr size_t MaxIntegerLength{32};
 
-void validateCapacity(size_t new_capacity) {
+uint64_t newCapacity(uint32_t existing_capacity, uint32_t size_to_append) {
```

Comment:
Nit: you could make `existing_capacity` a `uint64_t` to avoid the cast, but this doesn't really matter.